### ISSUE TITLE
No detection on Video with trained weights using our custom datasets

### ISSUE THREAD

**Issue by kkt0521** [AlexeyAB/darknet#2333]
We trained weigths with our custom datasets.
We made datasets with yolo_mark.
Training was fine and testing on image file was fine too.
But testing on video file, there were any results.

[Test on Image]
![predictions_retrain2_cap_2](https://user-images.githubusercontent.com/28796035/52095331-ea5a7700-2605-11e9-9aa8-ba2fa5dd326b.jpg)

[Test on Video]
![test_on_video](https://user-images.githubusercontent.com/28796035/52095575-d6fbdb80-2606-11e9-95e4-cef4891c69e4.JPG)

I think it can't find objects with trained weights.
Then, Why testing on image was done???

Please, notice me.

**Comment by tahsinsiad** [comment]
How many images do you use in your dataset for each class?

**Comment by AlexeyAB** [comment]
* Show command that you use for testing on images and video
* What cfg-file do you use?
* Show screenshot of video where it can't test anything

**Comment by tahsinsiad** [comment]
I also face same problem Sir. I have 2 classes. I train 3000 images( 1500 for car, 1500 for bike). Multiple car/bike detection is quite ok. But, not detect all the cars/bikes properly.  In case of video, it is very poor. Does this happen for lack of training images? 

I use `yolov3-tiny.cfg` and `yolov3-tiny.weights` for this.

I use `darknet.exe detector demo data/obj.data yolov3-tiny-obj.cfg backup/yolov3-tiny-obj_13000.weights traffic.avi -out_filename result.avi` for video detection.

**Comment by AlexeyAB** [comment]
@tahsinsiad 

> Does this happen for lack of training images?

Yes. For each object which you want to detect - there must be at least 1 similar object in the Training dataset.

https://github.com/AlexeyAB/darknet#how-to-improve-object-detection

> for each object which you want to detect - there must be at least 1 similar object in the Training dataset with about the same: shape, side of object, relative size, angle of rotation, tilt, illumination. So desirable that your training dataset include images with objects at diffrent: scales, rotations, lightings, from different sides, on different backgrounds - you should preferably have 2000 different images for each class or more, and you should train 2000*classes iterations or more

**Comment by tahsinsiad** [comment]
@AlexeyAB 

![car](https://user-images.githubusercontent.com/19798218/52164815-a3fa3a80-2721-11e9-8136-2c77f93e5a1c.jpg)

Here there are 4 cars. I want to annotate 4 cars. Are multiple annotations accepted?

**Comment by AlexeyAB** [comment]
@tahsinsiad 

> Here there are 4 cars. I want to annotate 4 cars. Are multiple annotations accepted?

Sure. You **must** annotate all 4 cars: https://github.com/AlexeyAB/darknet#how-to-improve-object-detection

> **check that each object are mandatory labeled in your dataset - no one object in your data set should not be without label.** In the most training issues - there are wrong labels in your dataset (got labels by using some conversion script, marked with a third-party tool, ...). Always check your dataset by using: https://github.com/AlexeyAB/Yolo_mark

**Comment by kkt0521** [comment]
I use yolo-obj.cfg file made by Yolo-Mark.

<yolo-obj.cfg>
[net]
# Testing
batch=1
subdivisions=1
# Training
#batch=256
#subdivisions=64
height=416
width=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.0001
max_batches = 45000
policy=steps
steps=100,25000,35000
scales=10,.1,.1

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=64
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=128
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky


#######

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=1024
activation=leaky

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=1024
activation=leaky

[route]
layers=-9

[reorg]
stride=2

[route]
layers=-1,-3

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=1024
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=45
activation=linear

[region]
anchors = 1.08,1.19,  3.42,4.41,  6.63,11.38,  9.42,5.11,  16.62,10.52
bias_match=1
classes=4
coords=4
num=5
softmax=1
jitter=.2
rescore=1

object_scale=5
noobject_scale=1
class_scale=1
coord_scale=1

absolute=1
thresh = .6
random=0
-------------------------------------------------------------------------------

I use the command like these.

(1)darknet.exe detector test data/obj_re.data yolo-obj.cfg yolo-obj_retrain_0123.weights data/obj/vv248.jpg

(2)darknet.exe detector demo data/obj_re.data yolo-obj.cfg yolo-obj_retrain_0123.weights test.mp4

I use two commands for testing image and video.

Is there any problem?
